### PR TITLE
Gridlayer opacity flicker fix

### LINKED ISSUE #3652: Opacity flickr when zooming WMS Layers
Hello, I have noticed a small flickr issue on layers which have an opacity lower than 1. 

If a layer has an opacity of 0.5 and you zoom in, it seems like the tiles are first loaded with opacity=1 and then are updated to opacity=0.5. This causes a weird flickr. This issue does not occur in leaflet 0.7.

Fiddle with leaflet 1.0-dev (zoom in and notice the flickr when the tiles are refreshing): http://jsfiddle.net/qf9huvk0/1/
Fiddle with leaflet 0.7.3 (no issues): http://jsfiddle.net/p6k4ejuq/3/

Is there any way I can solve this issue? Many thanks!

### PR DESCRIPTION
**Pull request by w8r**
#3652 

* Layer opacity applied on container
* Tile fade animation in 0...1 values

* IE<=9:
  * opacity set on tiles individually, no animation, as before

### CODE REVIEW

File: src/layer/tile/GridLayer.js [hunk 1]
```diff
@@ -156,14 +156,14 @@ L.GridLayer = L.Layer.extend({
 
 	_updateOpacity: function () {
 		if (!this._map) { return; }
-		var opacity = this.options.opacity;
 
 		// IE doesn't inherit filter opacity properly, so we're forced to set it on tiles
-		if (!L.Browser.ielt9 && !this._map._fadeAnimated) {
-			L.DomUtil.setOpacity(this._container, opacity);
+		if (L.Browser.ielt9 && !this._map._fadeAnimated) {
```
Comment (mourner) [inline src/layer/tile/GridLayer.js]: opacity in <IE9 is never animated, so the second condition is superfluous, right?

Comment (w8r) [inline src/layer/tile/GridLayer.js]: exactly, wanted to leave a note here as well

Comment (w8r) [inline src/layer/tile/GridLayer.js]: that was in the original code, I would change `&&` to `||` to make it work

Comment (mourner) [inline src/layer/tile/GridLayer.js]: Yeah, `||` sounds good.

**Comment by w8r** [comment]
Pushed that change as well

**Comment by mourner** [comment]
Squash into one commit and force-push please.

**Comment by w8r** [comment]
Done

**Comment by mourner** [comment]
Thanks!

**Pull request merged**